extra-cpp-bindings/src/lib.rs

```diff
@@ -75,24 +75,27 @@ mod ffi {
 
     /// The wallet registry entry
     pub struct WalletEntry {
+        /// wallet id
+        pub id: String,
         /// its name
         pub name: String,
         /// icon URLs
         pub image_url: ImageUrl,
-        /// native link (Android/Desktop), empty if none
-        pub native_link: String,
-        /// universal link (iOS), empty if none
-        pub universal_link: String,
+        /// mobile native link, empty if none
+        pub mobile_native_link: String,
+        /// mobile universal link, empty if none
+        pub mobile_universal_link: String,
+        /// desktop native link, empty if none
+        pub desktop_native_link: String,
+        /// desktop universal link, empty if none
+        pub desktop_universal_link: String,
     }
 
     /// The target platform
+    #[derive(Serialize, Deserialize, Clone, Debug)]
     pub enum Platform {
-        Android,
-        Ios,
-        Linux,
-        Mac,
-        Windows,
-        Browser,
+        Mobile,
+        Desktop,
     }
```

Comment:
why changing the platform to be more broad? I assume there may be e.g. Android-only wallets

Reply:
`App` are not that useful, sometimes it is confusing and unreliable. Some wallets not have desktop apps but still have desktop links, for example DeFi Wallet:

```
{
      "id": "f2436c67184f158d1beda5df53298ee84abfc367581e4505134b5bcf5f46697d",
      "name": "Crypto.com | DeFi Wallet",
      "description": "A non-custodial wallet that gives you access to a full suite of DeFi services in one place.",
      "homepage": "https://crypto.com/",
      "chains": [],
      "versions": [
        "1"
      ],
      "sdks": [
        "sign_v1"
      ],
      "app_type": "wallet",
      "image_id": "7c5ff577-a68d-49c5-02cd-3d83637b0b00",
      "image_url": {
        "sm": "https://registry.walletconnect.com/v2/logo/sm/7c5ff577-a68d-49c5-02cd-3d83637b0b00",
        "md": "https://registry.walletconnect.com/v2/logo/md/7c5ff577-a68d-49c5-02cd-3d83637b0b00",
        "lg": "https://registry.walletconnect.com/v2/logo/lg/7c5ff577-a68d-49c5-02cd-3d83637b0b00"
      },
      "app": {
        "browser": null,
        "ios": "https://apps.apple.com/us/app/id1262148500",
        "android": "https://play.google.com/store/apps/details?id=co.mona.android",
        "mac": null,
        "windows": null,
        "linux": null
      },
      "mobile": {
        "native": "cryptowallet:",
        "universal": "https://wallet.crypto.com"
      },
      "desktop": {
        "native": "cryptowallet:",
        "universal": null
      },
      "metadata": {
        "shortName": "Crypto.com",
        "colors": {
          "primary": "#1199fa",
          "secondary": null
        }
      }
    }
```


Actually, 
- For android, no need links.
- IOS only needs `desktop_universal_link` if it is not empty.
- Desktop only needs `desktop_native_link ` if it is not empty.